Packs/SymantecEDR/Integrations/SymantecEDR/SymantecEDR.py

```diff
@@ -0,0 +1,2734 @@
+                 is_fetch_comment: bool = False, fetch_status: list = None, fetch_priority: list = None,
+                 token: Optional[str] = None):
+
+        super().__init__(
+            base_url=base_url,
+            verify=verify,
+            proxy=proxy
+        )
+
+        self.client_key = client_id
+        self.secret_key = client_secret
+        self.first_fetch = first_fetch
+        self.fetch_limit = fetch_limit
+        self.is_incident_event = is_incident_event
+        self.is_fetch_comment = is_fetch_comment
+        self.fetch_status = fetch_status
+        self.fetch_priority = fetch_priority
+        self.access_token = token
+
+    @property
+    def headers(self) -> dict:
+        """
+        Client headers method
+        Returns:
+            self.headers
+        """
+        if self.access_token is None:  # for logging in, before self.access_token is set
+            return {'Content-Type': 'application/json'}
+
+        return {'Authorization': f'Bearer {self.access_token}', 'Content-Type': 'application/json'}
+
+    @staticmethod
+    def get_access_token_from_context(global_context: dict[str, Any]) -> str | None:
+        """
+        Symantec EDR on-premise get previous access token from global integration context
+        Args:
+            global_context(dict): Integration Context data
+        Returns:
+            return token or None
+        """
+        if save_timestamp := global_context.get('access_token_timestamp'):
+            now_timestamp = int(time.time())
+            if token := global_context.get('access_token'):
+                time_diff = int(now_timestamp - save_timestamp)
+                if time_diff <= SESSION_TIMEOUT_SEC:
+                    return token
+                else:
+                    LOG('Access token expired')
+        return None
+
+    def get_access_token_or_login(self) -> None:
+        """
+        Generate Access token
+        """
+        global_context = demisto.getIntegrationContext()
+        if last_access_token := self.get_access_token_from_context(global_context):
+            self.access_token = last_access_token
+            LOG("Access token still active. Re-use the same token")
+        else:
+            try:
+                response = self._http_request(
+                    method='POST',
+                    url_suffix='/atpapi/oauth2/tokens',
+                    auth=(self.client_key, self.secret_key),
+                    data={'grant_type': 'client_credentials'},
+                    resp_type='response'
+                )
+                response.raise_for_status()
+            except requests.exceptions.HTTPError as exc:
+                if error_msg := HTTP_ERRORS.get(exc.response.status_code):
+                    raise DemistoException(f'{error_msg}', res=exc.response) from exc
+                else:
+                    raise
```

Comment:
```suggestion
                raise
```

Reply:
changes done, removed else part and only raise in case of Exceptions.

Reply:
Done